dell/xps/13-9310/default.nix

```diff
@@ -0,0 +1,89 @@
+{ lib, pkgs, ... }:
+{
+  imports = [
+    ../../../common/cpu/intel
+    ../../../common/pc/laptop
+  ];
+
+  # TODO: upstream to NixOS/nixpkgs
+  nixpkgs.overlays = [(final: previous: {
+    qca6390-firmware = final.callPackage ./qca6390-firmware.nix {};
+  })];
+
+  hardware.firmware = lib.mkBefore [
+    # Necessary for audio.
+    # https://bbs.archlinux.org/viewtopic.php?pid=1933548#p1933548
+    pkgs.sof-firmware
+    # Firmware for the AX500 (wi-fi & bluetooth chip).
+    pkgs.qca6390-firmware
+  ];
+
+  # Confirmed necessary to get audio working as of 2020-11-13:
+  # https://bbs.archlinux.org/viewtopic.php?pid=1933643#p1933643
+  boot.extraModprobeConfig = ''
+    options snd-intel-dspcfg dsp_driver=1
+  '';
+
+  # Touchpad goes over i2c.
+  # Without this we get errors in dmesg on boot and hangs when shutting down.
+  boot.blacklistedKernelModules = [ "psmouse" ];
+
+  # TODO: Remove this once landed in kernel.
+  # Apply kernel patch for xps 9310 wifi bug.
+  # https://patchwork.kernel.org/project/linux-wireless/patch/1605121102-14352-1-git-send-email-kvalo@codeaurora.org/
+  boot.kernelPatches = [
+    # kvalo qca6390 patches
+    # https://git.kernel.org/pub/scm/linux/kernel/git/kvalo/ath.git/log/?h=ath11k-qca6390-bringup
+    {
+      name = "add-64-bit-check-before-reading-msi-high-addr";
+      patch = builtins.fetchurl "https://git.kernel.org/pub/scm/linux/kernel/git/kvalo/ath.git/patch/?id=065c9528cc508cfbf6e3399582df29f76f56163c";
```

Comment:
Please use `pkgs.fetchpatch` with a checksum here or it breaks NixOS flakes usage.